gobblin-config-management/gobblin-config-core/src/test/java/gobblin/config/common/impl/TestSingleLinkedListConfigKeyPath.java

```diff
@@ -0,0 +1,21 @@
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigKeyPath;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+@Test(groups = { "gobblin.config.common.impl" })
+
+public class TestSingleLinkedListConfigKeyPath {
+  
+  @Test
+  public void testAll(){
```

Comment:
Please checks for :
* ROOT.getOwnName()
* ROOT.getAbsolutePathString()
* getParent() for ROOT and non-root.
* equals() and hashCode() using, say, a HashSet of paths.